.github/workflows/_test_jax.yaml

```diff
@@ -58,22 +63,68 @@ jobs:
         shell: bash -x {0}
         run: |
           docker run --gpus all ${{ inputs.JAX_IMAGE }} test-jax.sh -b backend-independent | tee test-backend-independent.log
-          if [[ $(grep 'ERROR:' test-backend-independent.log | wc -l) > 0 ]]; then
-            exit 1
-          fi
 
       - name: GPU-specific tests
         shell: bash -x {0}
         run: |
           docker run --gpus all ${{ inputs.JAX_IMAGE }} test-jax.sh -b gpu | tee test-gpu.log
-          if [[ $(grep 'ERROR:' test-gpu.log | wc -l) > 0 ]]; then
-            exit 1
+
+      # - name: download existing artifact
+      #   uses: dawidd6/action-download-artifact@v2.27.0
+      #   with:
+      #     github_token: ${{secrets.GITHUB_TOKEN}}
+      #     run_id: 6047993737
+      #     name: jax-unit-test-logs-${{ matrix.GPU_ARCH }}        
+      #     path: .
+
+      - name: Process results and generate report
+        if: success() || failure()
+        shell: bash -x -e {0}
+        run: |
+          # bring in utility functions
+          source .github/workflows/scripts/to_json.sh
+
+          badge_label='${{ matrix.GPU_ARCH }} Unit'
+
+          errors=$(cat test-*.log | grep -c 'ERROR:' || true)
+          failed_tests=$(cat test-*.log | grep -c 'FAILED in' || true)
+          passed_tests=$(cat test-*.log | grep -c 'PASSED in' || true)
+          total_tests=$((failed_tests + passed_tests))
+
+          if [[ ${errors} > 0 ]] || [[ ${total_tests} == 0 ]]; then
+            badge_message='error'
+            badge_color=red
+            summary='JAX unit test on ${{ matrix.GPU_ARCH }} did not complete due to errors.'
+          else
+            badge_message="${passed_tests}/${total_tests} passed"
+            if [[ ${failed_tests} == 0 ]]; then
+              badge_color=brightgreen
+            else
+              badge_color=yellow
+            fi
+            summary="JAX unit test on ${{ matrix.GPU_ARCH }}: $badge_message"
           fi
 
-      - name: Upload test logs
+          to_json \
+            summary \
+            errors total_tests passed_tests failed_tests \
+            badge_label badge_color badge_message \
+          > sitrep.json
+
+          schemaVersion=1 \
+          label="${badge_label}" \
+          message="${badge_message}" \
+          color="${badge_color}" \
+          to_json schemaVersion label message color \
+          > ${{ env.BADGE_FILENAME_FULL }}
```

Comment:
What do you think about `tee`ing, here and for sitrep.json, instead of redirect? I feel like it might make troubleshooting easier

Reply:
I thought `shell: bash -x -e {0}` should already provide a similar experience. Is my understanding correct?